agent/taskresource/logrouter/logconfig_unix.go

```diff
@@ -0,0 +1,144 @@
+
+package logrouter
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	generator "github.com/awslabs/go-config-generator-for-fluentd-and-fluentbit"
+)
+
+const (
+	// LogRouterTypeFluentd is the type of a fluentd log router.
+	LogRouterTypeFluentd = "fluentd"
+
+	// LogRouterTypeFluentbit is the type of a fluentbit log router.
+	LogRouterTypeFluentbit = "fluentbit"
+
+	// socketInputNameFluentd is the name of the socket input plugin for fluentd.
+	socketInputNameFluentd = "unix"
+
+	// socketInputNameFluentbit is the name of the socket input plugin for fluentbit.
+	socketInputNameFluentbit = "forward"
+
+	// socketInputPathOptionFluentd is the key for specifying socket path for fluentd.
+	socketInputPathOptionFluentd = "path"
+
+	// socketInputPathOptionFluentbit is the key for specifying socket path for fluentbit.
+	socketInputPathOptionFluentbit = "unix_path"
+
+	// outputLogOptionKeyFluentd is the key for the log option that specifies output plugin name for fluentd.
+	outputLogOptionKeyFluentd = "@type"
+
+	// outputLogOptionKeyFluentbit is the key for the log option that specifies output plugin name for fluentbit.
+	outputLogOptionKeyFluentbit = "Name"
+
+	// includePatternKey is the key for include pattern.
+	includePatternKey = "include-pattern"
+
+	// excludePatternKey is the key for exclude pattern.
+	excludePatternKey = "exclude-pattern"
+
+	// socketPathFormat is the format for socket file path.
+	socketPathFormat = "/var/run/%s.sock"
+)
+
+// generateConfig generates a FluentConfig object that contains all necessary information to construct
+// a fluentd or fluentbit config file for the log router.
+func (logRouter *LogRouterResource) generateConfig() (generator.FluentConfig, error) {
+	config := generator.New()
+
+	// Specify log stream input, which is a unix socket that will be used for communication between log router container
+	// and other containers.
+	var inputName, inputPathOption string
+	if logRouter.logRouterType == LogRouterTypeFluentd {
+		inputName = socketInputNameFluentd
+		inputPathOption = socketInputPathOptionFluentd
+	} else {
+		inputName = socketInputNameFluentbit
+		inputPathOption = socketInputPathOptionFluentbit
+	}
+	config.AddInput(inputName, "", map[string]string{
+		inputPathOption: fmt.Sprintf(socketPathFormat, logRouter.logRouterType),
+	})
+
+	if logRouter.ecsMetadataEnabled {
+		// Add ecs metadata fields to the log stream.
+		config.AddFieldToRecord("ecs_cluster", logRouter.cluster, "*").
+			AddFieldToRecord("ecs_task_arn", logRouter.taskARN, "*").
+			AddFieldToRecord("ecs_task_definition", logRouter.taskDefinition, "*")
+		if logRouter.ec2InstanceID != "" {
+			config.AddFieldToRecord("ec2_instance_id", logRouter.ec2InstanceID, "*")
+		}
+	}
+
+	// Specify log stream output. Each container that uses the log router container to stream logs
+	// will have its own output section, with its own log options.
+	fields := strings.Split(logRouter.taskARN, "/")
```

Comment:
nit: the SDK go has an [arn package](https://docs.aws.amazon.com/sdk-for-go/api/aws/arn/)

(I'm fine with this though)

Reply:
i think even if we use that package we'd still have to split the "Resource" field in the same way.. so keeping this